google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/RowCell.java

```diff
@@ -62,7 +63,8 @@ public static RowCell create(
       long timestamp,
       @Nonnull List<String> labels,
       @Nonnull ByteString value) {
-    return new AutoValue_RowCell(family, qualifier, timestamp, value, labels);
+    // Ensure that the list is serializable and optimize for the common case
+    return new AutoValue_RowCell(family, qualifier, timestamp, value, ImmutableList.copyOf(labels));
```

Comment:
This doesnt optimize for the common case. The common case is empty labels. This is currently creating an extra allocation. Optimization would be:

labels = labels.isEmpty()? ImmutableList.of() : ImmutableList.copyOf(labels);

Reply:
Yes, indeed. I assumed `ImmutableList.of()` accepts var args, but it is not.
Ternary Operator does not work here(the compiler is not happy), so I have to use `if`.

Please take a look.